mate-panel/panel-run-dialog.c

```diff
@@ -888,20 +904,23 @@ panel_run_dialog_add_items_idle (PanelRunDialog *dialog)
 	for (l = all_applications; l; l = l->next) {
 		MateMenuTreeEntry *entry = l->data;
 		GtkTreeIter    iter;
-		const gchar *icon = matemenu_tree_entry_get_icon (entry);
-		GIcon *gicon = panel_gicon_from_icon_name (icon);
+		GDesktopAppInfo *ginfo;
+		GIcon *gicon = NULL;
+
+		ginfo = matemenu_tree_entry_get_app_info (entry);
+		gicon = g_app_info_get_icon(G_APP_INFO(ginfo));
 
 		gtk_list_store_append (dialog->program_list_store, &iter);
 		gtk_list_store_set (dialog->program_list_store, &iter,
 				    COLUMN_GICON,     gicon,
-				    COLUMN_NAME,      matemenu_tree_entry_get_display_name (entry),
-				    COLUMN_COMMENT,   matemenu_tree_entry_get_comment (entry),
-				    COLUMN_EXEC,      matemenu_tree_entry_get_exec (entry),
+				    COLUMN_NAME,      g_app_info_get_display_name(G_APP_INFO(ginfo)),
+				    COLUMN_COMMENT,   g_app_info_get_description(G_APP_INFO(ginfo)),
+				    COLUMN_EXEC,      g_app_info_get_commandline(G_APP_INFO(ginfo)),
 				    COLUMN_PATH,      matemenu_tree_entry_get_desktop_file_path (entry),
 				    COLUMN_VISIBLE,   TRUE,
 				    -1);
-
-		g_object_unref (gicon);
+		if (gicon != NULL)
```

Comment:
`g_app_info_get_icon` is "transfer none", which means the icon shouldn't be unref'ed.